src/main/java/io/ticktok/validator/ClockRequest.java

```diff
@@ -0,0 +1,32 @@
+package io.ticktok.validator;
+
+import org.apache.commons.lang3.StringUtils;
+
+import static io.ticktok.Ticktok.TicktokException;
+
+public class ClockRequest {
+
+    private String name;
+    private String schedule;
+
+    public ClockRequest(String name, String schedule) {
+        this.name = validate(name);
+        this.schedule = validate(schedule);
+    }
+
+    private String validate(String prop) {
```

Comment:
This is really poor validation and need to think throughout,
1. you should never throw exception on construction
2. the validation doesn't tell what is missing 
3. the exception is too general

Reply:
fixed